sdcm/data_operations/data_operation.py

```diff
@@ -0,0 +1,293 @@
+            try:
+                partition_values = self.get_unique_partition_values(partitions_amount=partitions_amount_to_delete)
+                where_condition = self.build_where_condition(columns=self.schema_helpers.partition_keys,
+                                                             values=partition_values)
+                max_clustering_values = self.get_max_clustering_value_in_partition(where_condition=where_condition)
+                if not max_clustering_values:
+                    return
+
+                where_condition_to_delete = []
+                for pk_clause, clustering_value in max_clustering_values.items():
+                    where_condition_to_delete.append(f" {pk_clause} and "
+                                                     f"{clustering_value['ck_name']} > {int(clustering_value['value'] / 2)}")
+
+                self.log.debug("Where condition for delete: %s", where_condition_to_delete)
+
+                self.delete_partition(where_condition_to_delete=where_condition_to_delete)
+                time.sleep(self.interval)
+                self.validate_deleted_rows_not_resurrected(where_condition_to_delete=where_condition_to_delete)
+                return
+            except Exception as error:  # pylint: disable=broad-except
+                self.log.error(traceback.format_exc())
+                case_event.severity = Severity.CRITICAL if isinstance(error, ResurrectionDataFound) else Severity.ERROR
+                case_event.message = str(error)
+                return
+
+
+# pylint: disable=too-many-instance-attributes
+class DataOperationThread(Thread):
+    """
+    Run thread with data manipulations during entire test. Sleep between operations.
+    For now one operation is supported: delete full partition
+    """
+    DEFAULT_INTERVAL_BETWEEN_OPERATIONS_SEC = 900
+
+    def __init__(self, tester, thread_name: str):
+        self.tester = tester
+        self.interval = self.DEFAULT_INTERVAL_BETWEEN_OPERATIONS_SEC
+        self.schema_helpers = SchemaHelpers(tester=self.tester)
+        self.data_operations = DataOperations(schema_helpers=self.schema_helpers, interval=self.interval)
+        # parameter "data_operation_ks_class" defined as:
+        #  - {'replication_factor': 3}
+        #  OR
+        #  - {'replication_factor': ['eu-westscylla_node_west': 3, 'us-west-2scylla_node_west': 2]}
+        ks_class = self.tester.params.get("data_operation_ks_class")
+        self.schema_helpers.create_default_keyspace_query(replication_factor=ks_class["replication_factor"])
+        self.duration_sec = self.tester.params.get('test_duration') * 60
+        self.log = logging.getLogger(self.__class__.__name__)
+        self.scenarios = [self.data_operations.delete_full_partition, self.data_operations.delete_half_partition]
+        # TODO: decrease gc_grace_seconds?
+        self.tester.run_stress(stress_cmd=self.schema_helpers.DEFAULT_STRESS_WRITE_COMMAND)
```

Comment:
we are not waiting for stress to complete, so when we create this class and run thread then 2 stress thread can start.
I think we could remove it as anyway thread loop will keep it running in case it's needed